src/color.rs

```diff
@@ -180,7 +182,12 @@ impl<T: Primitive + 'static> Pixel for $ident<T> {
 
     const COLOR_MODEL: &'static str = $interpretation;
 
-    const COLOR_TYPE: ColorType = $color_type;
+    fn color_type() -> ColorType {
+        match std::mem::size_of::<T>() {
+            1 => $color_type_u8,
+            _ => $color_type_u16,
+        }
+    }
```

Comment:
There might be a way to keep this an associated `const` despite the added logic. A very similar workaround is used within `std` and `core` in various places (and has a long C legacy).

```
const COLOR_TYPE = [$color_type_u8, $color_type_u16][(std::mem::size_of::<T>() > 1) as usize];
```